src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipeline.cs

```diff
@@ -51,8 +48,40 @@ public async Task<Response> SendRequestAsync(HttpPipelineRequest request, Cancel
             }
         }
 
-        HttpPipelineTransport Transport {
-            get => (HttpPipelineTransport)_pipeline.Span[_pipeline.Length - 1];
+        public static HttpPipeline Build(HttpPipelineOptions options, Action<IList<HttpPipelinePolicy>> configurePolicies)
```

Comment:
I really am not a big fan of actions/delegate based public APIs.

Reply:
I don't like this part too, the other alternative I had is to add a base type for the client  and have a virtual `HttpPipeline Build` method on it.

Reply:
And this is to avoid mutating the passed in options?

Reply:
Yes

Reply:
Also, the API limits client modifications to those that can be done through the ILists. This does not seem flexible enough. But, if it was, I would just make the build method take two collections (one for pre and one for after).

Reply:
The client is free to clear the list and redo it from scratch, seems flexible enough. 

> But, if it was, I would just make the build method take two collections (one for pre and one for after).

I'll do this.

Reply:
> I really am not a big fan of actions/delegate based public APIs.

This isn't germane to this particular review, I'm hijacking a bit.   I find myself with the opposite view; I'm a fan of higher order functions as a general design.  I'm curious as to your thoughts around this and very interested in the reasoning behind it.